app/lib/onboarding/pages/onboarding.dart

```diff
@@ -66,7 +69,11 @@ class OnboardingPage extends HookWidget {
     return TextButton(
       onPressed: () {
         if (isLastPage) {
-          context.router.replace(const LoginRouter());
+          if (_isLoggedIn) {
+            context.router.pop();
+          } else {
+            context.router.replace(LoginRouter());
+          }
```

Comment:
After the ` OnboardingPage`, you should never be logged in. So why are we handling that case here?